### PR TITLE
DD-1459 Fix uncontrolled data used in path expression

### PR DESCRIPTION
**Pull request by aliassheikh**
Fixes DD-1459 Fix Uncontrolled data used in path expression

# Description of changes
  - Added `getSecurePath(Path path)` to `ImportArea.java`
  - Checking the trusted base folders of `dd-ingest-flow` in the classes `ImoprtResource` and `MigrationResource`

# How to test and follow it in debugger
  - Starting vagrant - open terminal and follow the instructions below :
    - `start-preprovisioned-box.py -s dev_archaeology`
    - `deploy.py  -m dd-ingest-flow dev_archaeology`
    - `vagrant ssh`
    - `cd /var/opt/dans.knaw.nl/tmp`
    - `tree -L 5` (to oversee the working folders and deposits)
    -  Create to test folders inside import and migration folders
      - `mkdir import/deposits/test-folder`
      - `mkdir migration/deposits/test-folder`
    - copy one of the deposits into new created test-folders
      - `cp -r auto-ingest/outbox/processed/33afaff6-e08a-4bfa-9960-2b48b616d2c9/ import/inbox/test-folder/` 
      - `cp -r auto-ingest/outbox/processed/33afaff6-e08a-4bfa-9960-2b48b616d2c9/ migration/deposits/test-folder/`
    - open dd-ingest-flow porject in IntelliJ and run remote-debug with info:
      - remote debug
      - host: dev.sword2.archaeology.datastations.nl
      - port: 20302 
    - Commands to import or migrate deposits:
      - `ingest-flow start-import -s import/inbox/test-folder/33afaff6-e08a-4bfa-9960-2b48b616d2c9/`
      - `ingest-flow start-migration -s migration/deposits/test-folder/33afaff6-e08a-4bfa-9960-2b48b616d2c9/`

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/core-systems

### CODE REVIEW

**Review by jo-pol — COMMENTED** [review]
Just some style details

File: src/main/java/nl/knaw/dans/ingest/resources/ImportsResource.java [hunk 1]
```diff
@@ -47,7 +47,8 @@ public Response startImport(StartImport start) {
         log.debug("Received command = {}", start);
         String batchName;
         try {
-            batchName = importArea.startImport(start.getInputPath(), start.isBatch(), start.isContinue());
+            java.nio.file.Path securePath = importArea.getSecurePath(start.getInputPath());
```
[the remark this answers is not in the document]
Comment (janvanmansum) [reply src/main/java/nl/knaw/dans/ingest/resources/ImportsResource.java:50]: I agree, let's use `var` here.

File: src/main/java/nl/knaw/dans/ingest/resources/MigrationsResource.java [hunk 2]
```diff
@@ -47,7 +47,8 @@ public Response startImport(StartImport start) {
         log.info("Received command = {}", start);
         String taskName;
         try {
-            taskName = migrationArea.startImport(start.getInputPath(), start.isBatch(), start.isContinue());
+            java.nio.file.Path securePath = migrationArea.getSecurePath(start.getInputPath());
```
[the remark this answers is not in the document]
Comment (janvanmansum) [reply src/main/java/nl/knaw/dans/ingest/resources/MigrationsResource.java:50]: I agree, let's use `var` here.

**Review by janvanmansum — CHANGES_REQUESTED** [review]
See inline comments.

File: src/main/java/nl/knaw/dans/ingest/resources/ImportsResource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (aliassheikh) [reply src/main/java/nl/knaw/dans/ingest/resources/ImportsResource.java:50]: in imports there is already a `Path` type (`import javax.ws.rs.Path;`)
In my case, it is `java.nio.file.Path`. I think it is clearer to have it.
But you asked and I am going to change it to `var`

File: src/main/java/nl/knaw/dans/ingest/resources/MigrationsResource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (aliassheikh) [reply src/main/java/nl/knaw/dans/ingest/resources/MigrationsResource.java:50]: in imports there is already a `Path` type (`import javax.ws.rs.Path;`)
In my case, it is `java.nio.file.Path`. I think it is clearer to have it.
But you asked and I am going to change it to `var`

File: src/main/java/nl/knaw/dans/ingest/resources/ImportsResource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (janvanmansum) [reply src/main/java/nl/knaw/dans/ingest/resources/ImportsResource.java]: OK, I see your point. I prefer `var`, the function name `getSecurePath` already makes clear sufficiently what type of object is returned.

**Pull request merged**